src/parse.rs

```diff
@@ -99,116 +99,118 @@ impl AST {
     }
 }
 
-pub fn parse_project_context(root_path: &Path) -> std::io::Result<compat::JSSAContext> {
-    let path_str = root_path.to_str().unwrap_or("");
-    let modules = parse_directory(&root_path)?;
+#[derive(Debug, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct Directory {
+    files: Vec<PathBuf>,
+    sub_directories: Vec<Directory>,
+    path: PathBuf,
+}
+
+pub fn parse_project_context(directory: &Directory) -> std::io::Result<compat::JSSAContext> {
```

Comment:
Yeah. Those actually go away completely because of how `compat::JSSAContext` is structured. It's there just for the sake of invoking `.into()` later on.